DiceKit/FrequencyDistribution.swift

```diff
@@ -222,9 +239,25 @@ extension FrequencyDistribution {
         }
     }
     
-    // TODO: public func divide(x: FrequencyDistribution) -> FrequencyDistribution
-    // Needed to support power(-n)
-    
+    public func divide(y: FrequencyDistribution) -> FrequencyDistribution {
+        guard let initialK = orderedOutcomes.first, lastK = orderedOutcomes.last else {
+            return .additiveIdentity
+        }
+        guard let firstY = y.orderedOutcomes.first, lastY = y.orderedOutcomes.last, firstYFrequency = y[firstY] where firstYFrequency != 0.0 else {
+            fatalError("Divide by zero! :(")
+        }
+        var xFrequencies: FrequenciesPerOutcome = [:]
+        var p: Frequency
+        for k:Outcome in initialK...(lastK + lastY) {
+            p = 0.0
+            for (n, frequency) in xFrequencies {
+                p += frequency * (y[k - n] ?? 0)
+            }
+            xFrequencies[k - firstY] = ((frequenciesPerOutcome[k] ?? 0) - p) / firstYFrequency
+        }
+        let delta = ProbabilityMassConfig.probabilityEqualityDelta
+        return FrequencyDistribution(xFrequencies).filterZeroFrequencies(delta)
```

Comment:
What if I just got rid of the class function `filterZeroFrequencies` and replaced it with an overload init? Instead of passing `filterZeroFrequencies: Bool` we would just pass our delta there. Where we filter zero frequencies in our code now we would just replace with a line that makes a new frequency distribution object. In order to allow users to filter zero frequencies, we would make this new init public, instead of internal.